### PR TITLE
Design pattern behaviour

### PR DESCRIPTION
**Pull request by LMiljak**
(Re)-added tests.

<testing

### CODE REVIEW

**Comment by Taeir** [comment]
Tests look good. The travis build passed, and most other changes are because of renaming IBehaviour. Good to be merged.

<testing

File: src/test/java/com/github/fishio/behaviours/TestFrozenBehaviour.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	 * Tests the preMove method.
+	 */
+	@Test
+	public void preMove() {
+		behaviour.preMove();
+		
+		// The speedVector should not have changed as a result of calling the preMove method.
+		assertEquals(new Vec2d(0, 0), behaviour.getSpeedVector());
+	}
+
```
Comment (jessearens) [inline src/test/java/com/github/fishio/behaviours/TestFrozenBehaviour.java:44]: You should test the behaviour after the frozen behavior is over.

<testing

**Comment by jessearens** [comment]
Looks good, except for my one comment.

<testing

File: src/test/java/com/github/fishio/behaviours/TestFrozenBehaviour.java (hunk 1 above)
Comment (Taeir) [inline src/test/java/com/github/fishio/behaviours/TestFrozenBehaviour.java:44]: This is hard to do on travis due to delays, but I agree that this should be tested at a later date.

<testing

Comment (jessearens) [inline src/test/java/com/github/fishio/behaviours/TestFrozenBehaviour.java:44]: I agree that this doesn't have to be done before this request can be merged but it should be done some time.

<testing

Comment (LMiljak) [inline src/test/java/com/github/fishio/behaviours/TestFrozenBehaviour.java:44]: I know it's already merged, but I still have to diagree.

This test is written for the FrozenBehaviour class, but the FrozenBehaviour class does not have a "duration", it has nothing to do with time. What you're confused with is the PuFreeze (Freeze powerup) which sets the behaviour of enemyfish to the FrozenBehaviour, then after a certain duration resets it back to its old behaviour. 
Those tests can be found in the TestPuFreeze class (testStartEffect and testEndEffect). startEffect and endEffect are called at the right moment by PuFreeze's superclass: DurationPowerUp, but that is already tested!

<testing

Comment (jessearens) [inline src/test/java/com/github/fishio/behaviours/TestFrozenBehaviour.java:44]: You're completely right, I was mistaken with the Freeze power up, which does have it's links with this behaviour class but is not the same. However, as #136 suggest, the tests in TestpuFreeze do not really work the way they should as they should test whether the resulting behaviour after freeze is the same behaviour as before freeze (which should be the case, but this is not so).

<testing

**Pull request merged**